### PR TITLE
Catch-up timer

### PR DESCRIPTION
**Pull request by metakirby5**
Implement catch-up behavior into server game loop using timer. That way, we don't have to worry about network stuff causing our updates to fall behind.

Also provide a timer to client for timed callbacks. This makes sense now that timer does not actually block.

### CODE REVIEW

File: src/aldente_client.cpp [hunk 1]
```diff
@@ -101,13 +102,18 @@ void AldenteClient::start() {
 
     std::cerr << "Starting client..." << std::endl;
 
+    // Used for callbacks
+    Timer timer(GAME_TICK);
```
Comment (richard1) [inline src/aldente_client.cpp:106]: Just confirming, so now the client's clock is the same duration as the server's?

**Review by richard1 — APPROVED** [review]
LGTM

File: src/aldente_client.cpp (hunk 1 above)
Comment (metakirby5) [reply src/aldente_client.cpp:106]: Since the catch up is non blocking, this essentially defines the quantization of timed callbacks to be 30ms.

**Comment by metakirby5** [comment]
@jytang if you wanted to see this

**Pull request merged**